Doc: Add breaking changes (#13376)
elastic/logstash: docs/static/breaking-changes.asciidoc

// tag::notable-breaking-changes[]
[discrete]
[[bc-java-11-minimum]]
===== Java 11 minimum
Logstash requires Java 11 or later.
By default, Logstash will run with the bundled JDK, which has been verified to
work with each specific version of Logstash, and generally provides the best
performance and reliability.

[discrete]
[[bc-java-home]]
===== Support for JAVA_HOME removed
We've removed support for using `JAVA_HOME` to override the path to the JDK. 

> Users should set the value of `LS_JAVA_HOME` to the path of their preferred JDK
> if they need to use a version other than the bundled JDK. 

Reviewer: I would prefer we _lead_ with the qualification of who should care about this. Most users do not have a legitimate reason to provide their own JDK.

```suggestion
Users who need to use a version other than the bundled JDK should set the value
of `LS_JAVA_HOME` to the path of their preferred JDK. 
```

Author: Love it!  ❤️
